lib/ansible/module_utils/basic.py

```diff
@@ -2031,71 +2043,76 @@ def preexec():
             if before_communicate_callback:
                 before_communicate_callback(cmd)
 
-            # the communication logic here is essentially taken from that
-            # of the _communicate() function in ssh.py
-
             stdout = b''
             stderr = b''
-            try:
-                selector = selectors.DefaultSelector()
-            except (IOError, OSError):
-                # Failed to detect default selector for the given platform
-                # Select PollSelector which is supported by major platforms
+
+            # Mirror the CPython subprocess logic and preference for the selector to use.
+            # poll/select have the advantage of not requiring any extra file
+            # descriptor, contrarily to epoll/kqueue (also, they require a single
+            # syscall).
```

Comment:
Also, select/poll are more efficient for a small amount of descriptors, kqueue/epoll are designed to scale a lot better, but this is mostly for a service listening for network connections.

Reply:
Except for the first line, the comment comes from the code of `subprocess` itself.  Just an FYI.

Reply:
even select/poll choice when dealing with stdin/out is really almost irrelevant.  select is O(highest file descriptor), while poll is O(num of descriptors).